woocommerce/woocommerce-ios
# Local notification: Notification to purchase plan 6 hour after trial subscription

Closes: #10264 


## Description

This PR adds a new scenario for local notification to remind users to purchase a paid plan 6 hours after their trial subscription.

Since we are replacing notification scenario 5 with scenario 7, this notification uses the same wording as scenario 6.

Since the logic for scheduling post-trial subscriptions is very similar for scenarios 6 and 7, I added some refactoring to reuse the logic with a method named `schedulePostSubscriptionNotification`. We just need to specify the scenario and the time interval for scheduling the notifications and the rest are taken care of.

## Testing instructions

- Build and run the app on a physical device for easy testing.
- Create a new store with a free trial subscription.
- When the store creation completes, notice in Xcode console: `🔵 Tracked local_notification_scheduled, properties: [AnyHashable("blog_id"): <site_id>, AnyHashable("is_iap_available"): true | false, AnyHashable("type"): "six_hours_after_free_trial_subscribed<site_id>", AnyHashable("is_wpcom_store"): true]`.
- Open the device Settings app to switch to 5h58m after the time of the store creation.
- After 2 minutes, a notification asking for purchasing a plan should be displayed.
- Tap the notification, the app should be opened. Notice in Xcode console: `🔵 Tracked local_notification_tapped, properties: [AnyHashable("type"): "six_hours_after_free_trial_subscribed<site_id>", AnyHashable("is_wpcom_store"): true, AnyHashable("is_iap_available"): true | false, AnyHashable("blog_id"):<site_id>]`.
- The app should navigate to IAP if you are testing with US storefront, or the plan detail screen otherwise.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/3f6df7ed-8e85-4d3b-9055-0a3d354c05ad" width=320 />

---
